src/app/modules/time-clock/pages/time-clock.component.ts

```diff
@@ -20,22 +23,31 @@ export class TimeClockComponent implements OnInit, OnDestroy {
   areFieldsVisible = false;
   activeTimeEntry: Entry;
   clockOutSubscription: Subscription;
+  storeSubscribe : Subscription;
+  storeSubject;
+  exponentialGrowth;
+
 
 
   constructor(
+    private featureManagerService: FeatureManagerService,
     private azureAdB2CService: AzureAdB2CService,
     private store: Store<Entry>,
     private toastrService: ToastrService,
     private actionsSubject$: ActionsSubject
   ) {}
 
-  ngOnDestroy(): void {
-    this.clockOutSubscription.unsubscribe();
-  }
+  
+
+  async ngOnInit(): Promise<void> {
```

Comment:
Just a suggestion, because I know you tested your solution and you can proceed with that. I would reconsider the use of async-await, I think  we could consume the Observable that return the FeatureManager service inside a function and depending of condition value, we could run or not-run the  logic to avoid exponential-growth.